core/common/src/main/java/alluxio/security/authentication/SaslStreamClientDriver.java

```diff
@@ -89,7 +91,7 @@ public void start(String channelId) throws UnauthenticatedException, Unavailable
       // Send the server initial message.
       mRequestObserver.onNext(mSaslHandshakeClientHandler.getInitialMessage(channelId));
       // Wait until authentication status changes.
-      mAuthenticated.get();
+      mAuthenticated.get(2 , TimeUnit.SECONDS);
```

Comment:
Should this be configurable, is 2 seconds a reasonable time in all cases?

Reply:
A discussion that @ggezer and I should probably have. I can remove this change for now and we can discuss later. Though I think it should have a timeout or else we will wait indefinitely for an authentication response.

Reply:
I believe making this equal to channel shutdown timeout would make sense. Probably need a separate key though.

Reply:
Update: The tests won't pass if there isn't a timeout so I'm going to address this here.

How about we do `alluxio.master.grpc.channel.authentication.timeout`. We can set the default to `5sec` (This is much less than the current `60sec` timeout for channel shutdown, but 60 seconds seems excessive). The key will be under `MASTER_GRPC_CHANNEL_AUTH_TIMEOUT`.

Thoughts?

Reply:
I'm okay with putting it at 30. Both of the parameters are lowered during integration tests anyways to decrease run time. See 4dc2335c4cb7af51ecf52e36cd49d8a0e3245d0b for the changes